pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -273,6 +275,22 @@ func fillDefaults(opr *operatorv1.LogStorage) {
 	if opr.Spec.Nodes == nil {
 		opr.Spec.Nodes = &operatorv1.Nodes{Count: 1}
 	}
+
+	if opr.Spec.ComponentResources == nil {
+		limits := corev1.ResourceList{}
+		requests := corev1.ResourceList{}
+		limits[corev1.ResourceMemory] = resource.MustParse(defaultEckOperatorMemorySetting)
+		requests[corev1.ResourceMemory] = resource.MustParse(defaultEckOperatorMemorySetting)
+		opr.Spec.ComponentResources = []operatorv1.ComponentResource{
+			{
+				ComponentName: operatorv1.ComponentNameECKOperator,
+				ResourceRequirements: &corev1.ResourceRequirements{
+					Limits:   limits,
+					Requests: requests,
+				},
+			},
+		}
+	}
```

Comment:
Not in fillDefaults here but I think where we call fillDefaults you should add a new function to verify the LogStorage spec and that should include that ComponentResources only has a ComponentName of ECKOperator.